validphys2/src/validphys/photon/compute.py

```diff
@@ -0,0 +1,265 @@
+        start_time = time.perf_counter()
+        photon_100GeV = np.array([self.lux[id].EvaluatePhoton(x, self.q_in2).total for x in self.xgrid])
+        photon_100GeV += self.generate_errors()
+        photon_100GeV /= self.xgrid
+        print("Time to compute photon:", time.perf_counter() - start_time)
+        # TODO : the different x points could be even computed in parallel
+
+        # Load eko and reshape it
+        with EKO.edit(self.path_to_eko_photon) as eko:
+            # If we make sure that the grid of the precomputed EKO is the same of 
+            # self.xgrid then we don't need to reshape
+            xgrid_reshape(eko, targetgrid = XGrid(self.xgrid), inputgrid = XGrid(self.xgrid))
+
+            # construct PDFs
+            pdfs = np.zeros((len(eko.rotations.inputpids), len(self.xgrid)))
+            for j, pid in enumerate(eko.rotations.inputpids):
+                if pid == 22 :
+                    pdfs[j] = photon_100GeV
+                    ph_id = j
+                if not self.qcd_pdfs[id].hasFlavor(pid):
+                    continue
+                pdfs[j] = np.array(
+                    [
+                        self.qcd_pdfs[id].xfxQ2(pid, x, self.q_in2) / x
+                        for x in self.xgrid
+                    ]
+                )
+            
+            # Apply EKO to PDFs
+            q2 = eko.mu2grid[0]
+            with eko.operator(q2) as elem:
+                pdf_final = np.einsum("ajbk,bk", elem.operator, pdfs)
+                # error_final = np.einsum("ajbk,bk", elem.error, pdfs)
+
+        photon_Q0 = pdf_final[ph_id]
+
+        # we want x * gamma(x)
+        return self.xgrid * photon_Q0
+    
+    def produce_interpolators(self):
+        self.photons_array = [self.compute_photon_array(i) for i in range(len(self.replicas_id))]
+        self.interpolator = [interp1d(self.xgrid, photon_array, fill_value=0., kind='cubic') for photon_array in self.photons_array]
+    
+    def compute(self, xgrid):
+        """
+        Compute the photon interpolating the values of self.photon_array.
+
+        Parameters
+        ----------
+        xgrid : nd.array
+            array of x values with shape (1,xgrid,1)
+        
+        Returns
+        -------
+        photon values : nd.array
+            array of photon values with shape (1,xgrid,1)
+        """
+        return [
+            self.interpolator[id](xgrid[0,:,0])[np.newaxis,:,np.newaxis]
+            for id in range(len(self.replicas_id))
+        ]
+    
+    def integrate(self):
+        """Compute the integral of the photon on the x range."""
+        return [trapezoid(self.photons_array[id], self.xgrid) for id in range(len(self.replicas_id))]
+    
+    def generate_error_matrix(self):
+        """generate error matrix to be used for the additional errors."""
+        if not self.fiatlux_runcard["additional_errors"] :
+            return None
+        extra_set = lhapdf.mkPDFs("LUXqed17_plus_PDF4LHC15_nnlo_100")
```

Comment:
This should not be hardcoded and in any event it should be loaded using the nnpdf tools rather than lhapdf.

Reply:
(e.g.. the grid would be downloaded if the `--net` option is on and we would get a nice error message rather than a hard crash if it fails to load)